### PR TITLE
[#36] Implement custom error handling in user-microservice

### LINKED ISSUE #36: Implement custom error handling in user-microservice

### PR DESCRIPTION
**Pull request by MaksimPegov**

### CODE REVIEW

File: users/src/main/java/com/maksimpegov/users/UsersController.java [hunk 1]
```diff
@@ -22,28 +24,56 @@ public UsersController(UsersService usersService) {
...
+        HttpStatus httpStatus;
+        try {
+            httpStatus = HttpStatus.valueOf(status);
+        } catch (NumberFormatException e) {
+            httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+        }
+        return httpStatus;
+    }
+
+    public UserDto mapUser(User user) {
```
Comment (antonpegov) [inline users/src/main/java/com/maksimpegov/users/UsersController.java]: Why you use 'public' here? Are you going to call this method from outside the controller?

File: users/src/main/java/com/maksimpegov/users/UsersController.java [hunk 2]
```diff
@@ -22,28 +24,56 @@ public UsersController(UsersService usersService) {
...
+        return mapper.userToUserDto(user);
+    }
+
+    public UserInfo mapToInfo(User user) {
```
Comment (antonpegov) [inline users/src/main/java/com/maksimpegov/users/UsersController.java]: UserInfo is also DTO?

File: users/src/main/java/com/maksimpegov/users/exeption/ApiException.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.maksimpegov.users.exeption;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.time.ZonedDateTime;
+
+@Getter
+@AllArgsConstructor
+public class ApiException {
```
Comment (antonpegov) [inline users/src/main/java/com/maksimpegov/users/exeption/ApiException.java:10]: Why you creating two exceptopns classes?

**Review by antonpegov — APPROVED** [review]

File: users/src/main/java/com/maksimpegov/users/UsersController.java (hunk 1 above)
Comment (MaksimPegov) [reply users/src/main/java/com/maksimpegov/users/UsersController.java]: yep, i will change it, my fault

File: users/src/main/java/com/maksimpegov/users/exeption/ApiException.java (hunk 3 above)
Comment (MaksimPegov) [reply users/src/main/java/com/maksimpegov/users/exeption/ApiException.java:10]: ApiException is exception that my service will return, and ApiRuntimeException is exception that can be thrown in code, it extends RuntimeException(Java exception that cam be thrown at runtime)

**Pull request merged**